src/components/shop/menu.js

```diff
@@ -5,23 +5,31 @@
  * @license Apache-2.0
  */
 import { createMenuButtons } from "./menu-buttons.js";
-import { setVisible, resize, getHalfRectBounds, getInnerRectBounds, createRect, getSafeArea } from "./shop-layout.js";
+import {
+    setVisible,
+    resize,
+    getInnerRectBounds,
+    createRect,
+    getSafeArea,
+    createPaneBackground,
+} from "./shop-layout.js";
 
 export const createMenu = scene => {
     const bounds = getSafeArea(scene.layout);
-    const { buttonsRight } = scene.config.menu;
 
     const container = scene.add.container();
     container.config = scene.config;
     container.setVisible = setVisible(container);
     container.resize = resize(container);
-    container.memoisedBounds = bounds; // workaround, pending CGPROD-2887
+    container.memoisedBounds = bounds;
 
-    container.add([
-        createRect(scene, getHalfRectBounds(bounds, !buttonsRight), 0xff0000),
-        createRect(scene, getHalfRectBounds(bounds, buttonsRight), 0xff00ff),
+    const contents = [
         createRect(scene, getInnerRectBounds(scene), 0x0000ff),
-    ]);
+        // createRect(scene, bounds, 0x000000),
+        createPaneBackground(scene, bounds, "menu"),
```

Comment:
so are you saying if there is no key we cannot omit this call? what is relying on it

Reply:
We've got a container that keeps resizing to the safe area. The menu and confirm panes then use that container's bounds to do their layout. The rect on line 27 is basically the button positions. The createPaneBackground call on line 29 has to return some gameObject that's sized to the safe area. (The preferred approach would have been to have the menu and confirm pages get the safe area directly and do layout based on that- however, that fell foul of Genie's debouncing of resize events, which, if I'm remembering correctly, created a race condition.)

Reply:
guess the lesson here is 'ugh containers'